src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -1,33 +1,35 @@
-package seedu.address.model.expense;
+package seedu.address.model.transaction;
 
 import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
-import java.time.LocalDateTime;
 import java.util.Objects;
 
 import seedu.address.commons.util.ToStringBuilder;
 
 /**
- * Represents an Expense in the finance tracker.
+ * Represents a Transaction in the finance tracker.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
-public class Expense {
+public class Transaction {
     private final Name name;
     private final Amount amount;
     private final Category category;
-    private final LocalDateTime dateTime;
+    private final DateTime dateTime;
     private final Location location;
+    private final Type type;
 
     /**
      * Every field must be present and not null.
      */
-    public Expense(Name name, Amount amount, Category category, LocalDateTime dateTime, Location location) {
-        requireAllNonNull(name, amount, category, dateTime, location);
+    public Transaction(Name name, Amount amount, Category category, DateTime dateTime,
+                       Location location, Type type) {
+        requireAllNonNull(name, amount, category, dateTime, location, type);
```

Comment:
Should we remove the requireAllNonNull requirement or add another constructor since Location is optional?